File: LeetSync/2002-stone-game-viii/stone-game-viii.cpp
class Solution {
public:
    int stoneGameVIII(vector<int>& stones) {
        int n = stones.size();
        
        long long pref_sum = 0;
        for (int stone : stones) {
            pref_sum += stone;
        }
        long long dp = pref_sum;

        for (int i = n - 1; i >= 2; --i) {
            pref_sum -= stones[i];
            dp = max(dp, pref_sum - dp);
        }

        return dp;
    }
};

File: LeetSync/2002-stone-game-viii/stone-game-viii.py
class Solution:
    def stoneGameVIII(self, stones: List[int]) -> int:
        n = len(stones)
        
        pref_sum = sum(stones)
        
        dp = pref_sum
        
        for i in range(n - 1, 1, -1):
            pref_sum -= stones[i]
            dp = max(dp, pref_sum - dp)
            
        return dp